Develop

Zaknarfen opened the pull request.

Zaknarfen committed:
Refactoring code and structure

	- Moved the force service to the core module
	- Moved all PrimeNG references to the shared module
	- Refactored the shared module imports for other
	modules
	- Added two fields to the team member model
	- Changed the way that the contribute teams page load
	all the images, before we had to wait for requests and
	now we make all the requests in parallel
	- Related to #375

Zaknarfen committed:
Making a better contrib page loading

	- Changed the way the images are loaded in the contrib
	page, now they load async
	- Allowed the contributing teams page to gradually render
	images that are downloaded, and using a placeholder while
	waiting
	- Related to #375

Zaknarfen committed:
Improving code refactoring

	- Fixed a resizing error in the genes list component
	- Moved more methods to the api service
	- Changed calls in several components to use the new api
	methods
	- Moved all the loading logic into the a new data service
	method
	- Added parallel loading to the data service when loading
	chart data

Zaknarfen committed:
Fixing small issues

	- Closes #380
	- Closes #381

Commit:
Update README.md

Change badge to travis-ci.org

Zaknarfen committed:
Fixing style issues

	- Removed justification from text in the intro
	page
	- Changed modals look to match the Figma designs
	- Closes #382
	- Closes #383

Zaknarfen committed:
Merge pull request #404 from Sage-Bionetworks/kdaily-patch-1

Update README.md

Zaknarfen committed:
Fixing more styling issues

	- Fixed wrong colors and icon position in the
	multiselect component
	- Fixed wrong colors and icon position in the
	checkbox component
	- Added new images to fix design issues
	- Added an image for the nominated gene button,
	synapse account creation
	- Added a plus and minus icons
	- Fixed an issue with mobile modals
	- Closes #386
	- Closes #388
	- Closes #389
	- Closes #391

diegotoptal committed:
network widget fixes

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Fixing more styling issues

diegotoptal committed:
disable median chart interactivity

Zaknarfen committed:
Fixing Sprint 6 issues

	- Fixed a filter problem with the selectMenu
	chart when loading the same gene twice
	- Fixed left space clipping values in the forest
	plot
	- Added the ability to load the same gene from list
	or search without reloading the data
	- Changed the gene overview tests to reflect the new
	functionalities
	- Changed the gene service and gene service stub to
	reflect the changes
	- Closes #395
	- Closes #403
	- Closes #406

Zaknarfen committed:
More fixing to the Sprint 6 issues

	- Moved icons to the icons asset folder
	- Changed the default checked icon to the image
	used in the Figma design
	- Fixed the incorrect Y padding with the box
	plot chart
	- Added a link from Druggability in the Summary of
	Evidence to the Next Steps section of the page
	- Added a new package, ngx-page-scroll
	- Closes #381
	- Closes #401

diegotoptal committed:
small fix

zaro0508 committed:
Docker deployments to dockerhub

* Setup automated docker container deployments to the official
Sage dockerhub repository at
https://hub.docker.com/u/sagebionetworks/dashboard/
* Split CI builds into separate stages
* Setup Agora tests to run against the built and deployed agora
containers in dockerhub 'sagebionetworks' org.  This validates
the container builds and deployments.

Zaknarfen committed:
Fixes same gene reloading issue

	- Fixes a bug when reloading the same gene using the
	browser history button
	- Closes #403

Zaknarfen committed:
Fixing loading issue

	- Fixed a loading issue preventing information to get
	rendered properly
	- Changed a few api endpoints return values that were
	faulty
	- Changed a few design things related to the network widget
	- Fixed the column position for the network widget when
	there is no expression data
	- Closes #414

Zaknarfen committed:
Merge pull request #427 from zaro0508/container-ci-deploy

Docker deployments to dockerhub

diegotoptal committed:
fix network issues

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Docker deployments to dockerhub

zaro0508 committed:
setup test stage

Create a separate CI stage for tests.

Zaknarfen committed:
Adding more tests to the application

	- Added basic tests to several components

Zaknarfen committed:
Moving ci:travis out of the Dockerfile

zaro0508 committed:
cleanup CI build files

Zaknarfen committed:
Fixing November Milestone Issues

	- Removes the end of the text after RefSeq, or
	variances in the genes description
	- Added a inner section divider so prevent
	user confusion
	- Fixed an incorrect text display for the
	nomination of a target gene in the gene network
	component
	- Added visual feedback when searching genes
	- Added a new response interface type
	GeneInfosResponse
	- Changed the info icon to use the new design
	in Fimga
	- Fixed an issue with the BETA text not being
	bold
	- Updated vulnerable packages
	- Bumped data version to 8
	- Closes #412
	- Closes #413
	- Closes #425
	- Closes #428
	- Closes #429
	- Related to #433

zaro0508 committed:
tune nginx for AWS

Starting nginx container in a t2.large EC2 would result in error

  "20000 worker_connections exceed open file resource limit: 1024"

Update nginx configs to allow it to startup without error.

Zaknarfen committed:
Adding alphabetical order to the teams page

	- The teams page now shows teams in alphabetical
	order
	- Capitilized the Y axis label in the box-plot
	- Related to #375
	- Closes #395
	- Closes #435

diegotoptal committed:
similar page wip

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Adding alphabetical order to the teams page
  tune nginx for AWS

# Conflicts:
#	src/app/genes/gene-details/gene-similar/gene-similar.component.ts

Zaknarfen committed:
Hotfix for teams page

	- Added a hotfix to the teams ordering, images were correct
	but teams weren't

diegotoptal committed:
similar genes

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Hotfix for teams page

diegotoptal committed:
compile issues fix

zaro0508 committed:
Refactor mongodb containers

There is no need to have a container just to start the mongodb
service. Refactor to start mongo service directly from the
mongodb container.

Reference:
https://success.docker.com/article/how-can-i-access-mongodb-container-from-another-container

diegotoptal committed:
changed params to query test

zaro0508 committed:
Fix deploy container build

The part that builds the container was failing with message..
 Error: Failed to make Github request, rate limit reached

Disable gecko as suggested in webdriver-manager issue[1]

[1] https://github.com/angular/webdriver-manager/issues/307#issuecomment-377684918

diegotoptal committed:
remove private

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Fix deploy container build

diegotoptal committed:
fix some build issues

Zaknarfen committed:
Refactoring dialogs and fixing a minor issue

	- Changed the search spinner color
	- Added a new dialogs module
	- Moved the dialogs into a separate module
	- Created a more info component so that several
	components could use it to call dialogs preventing
	code duplication
	- Added tests to all new dialogs components
	- Related to #375
	- Related to #413

Zaknarfen committed:
Improving code

	- Reduced the bundle size
	- Removed unused code from third-party libraries
	- Added a module concatenation plugin to the
	Webpack build chain
	- Added the druggability component to the overview
	component
	- Related to #411

diegotoptal committed:
cleanup nd fixes

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Refactoring dialogs and fixing a minor issue
  Refactor mongodb containers

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Improving code

diegotoptal committed:
fix network selection and filtering

diegotoptal committed:
druggability info for nominated targets

diegotoptal committed:
fixes

diegotoptal committed:
text update

diegotoptal committed:
druggability and small fixes

diegotoptal committed:
druggability modal links

Zaknarfen committed:
Fixing styling issues

	- Changed Nominated Target list search to match
	the home page search
	- Removed Gene not listed from below the
	search bar
	- Fixed the search hint text
	- Updated video in the intro page
	- Added a placeholder image to hide the
	ugly YouTube player full of controls
	- Closes #413
	- Closes #439
	- Closes #440
	- Closes #441
	- Closes #444
	- Related to #446

Zaknarfen committed:
Quick fix

	- Changed hero background to cover

diegotoptal committed:
similar and brain regions open in new page

Zaknarfen committed:
UI fixes and changing scrolling

	- Added tab menu to the overview component
	- Fixed the hero background to look like the Figma
	designs
	- Fixed paddings in the overview header section

diegotoptal committed:
fixes

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  UI fixes and changing scrolling

diegotoptal committed:
nominated target form

Zaknarfen committed:
UI changes to the new Figma designs

	- Made navbar sticky
	- Moved intro video a bit up
	- Fixed background images
	- Changed genes module to handle the new UI logic
	- Added a new module for the gene overview
	- Closes #458
	- Closes #459
	- Closes #462

Zaknarfen committed:
Fixing route error

Zaknarfen committed:
Fixing relative paths errors

Zaknarfen committed:
Fixing errors

Zaknarfen committed:
Fixing relative path to gene routes

Zaknarfen committed:
Changing the final relative paths

diegotoptal committed:
nominated target modal and other fixes

diegotoptal committed:
fixed info icon color

diegotoptal committed:
fix Team photos are incorrectly assigned

Zaknarfen committed:
Continuing UI migration to new designs

	- Added a new component, rna.component
	- Added a new tab menu to be the second tier menu
	for the evidence tab
	- Moved some files from the genes modules into the overview
	module
	- Fixed some styling issues

diegotoptal committed:
create account updates

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Continuing UI migration to new designs

Zaknarfen committed:
Adding more pieces of the new UI design

	- Moved drugabbility component to be a child of the
	overview tab menu component
	- Added a new table for the Summary of Evidence
	- Added the final pieces for the overview tab menu
	- Commented the second tab menu for now

Zaknarfen committed:
Fixing invalid character error

diegotoptal committed:
Rna Modal and fixes

Zaknarfen committed:
More improvements to new UI design

	- Added more things from Figma's new design to the
	application
	- Added the druggability component to the tab menu with
	all needed parts
	- Finished three of the four tabs in the overview menu
	component

Zaknarfen committed:
Adding quick fix for the server db connection

diegotoptal committed:
similar page dialog

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Adding quick fix for the server db connection
  More improvements to new UI design
  Fixing invalid character error
  Adding more pieces of the new UI design

# Conflicts:
#	src/app/dialogs/nt-dialog/nt-dialog.component.html
#	src/app/genes/gene-overview/evidence-menu/rna/rna.component.html

diegotoptal committed:
test and fixes

diegotoptal committed:
text fix

diegotoptal committed:
replaces apostrophes with html coding

Zaknarfen committed:
Fixing UI issues

- Fixed heading sizes
- Changed box-plot background color
- Added the nominated link for the nominated gene
teams page
- Updated the Summary of Evidence modal content
- Fixed zoom problem when rendering genes list in
a small height window
- Changed all headers and gene names to the correct
colors
- Closes #450
- Closes #484
- Closes #514
- Closes #532

diegotoptal committed:
remove merging issue

zaro0508 committed:
fix retrieving params from SSM

SSM params are returned in alpha order so need to reference them
accordingly.

diegotoptal committed:
small fixes

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  fix retrieving params from SSM

diegotoptal committed:
small fixes

Zaknarfen committed:
Quick fix for tslint

Zaknarfen committed:
Quick fix for row chart

diegotoptal committed:
mobile menu wip

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Quick fix for row chart
  Quick fix for tslint
  Fixing UI issues

# Conflicts:
#	src/app/genes/genes-list/genes-list.component.html
#	src/app/genes/genes-list/genes-list.component.scss

diegotoptal committed:
small fix

diegotoptal committed:
fix network widget height

Zaknarfen committed:
Fixing the charts error

diegotoptal committed:
mobile fixes

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Fixing the charts error

Zaknarfen committed:
Attempt to debug chart inconsistency

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Attempt to debug chart inconsistency

zaro0508 committed:
setup CD to AWS Beanstalk

AWS Beanstalk is now setup to run Agora & Mongo[1].  Now we can auto
deploy a new version of the Agora app on every code change.

This setup also allows promoting code changes from Agora develop ->
staging and then eventually prod environments.

[1] http://agora-develop.ampadscicomp.org

diegotoptal committed:
fixes

diegotoptal committed:
fix

diegotoptal committed:
small fixes

gatof81 committed:
Merge pull request #542 from zaro0508/setup-ci

setup CD to AWS Beanstalk

zaro0508 commented:
change 0cdf16d9 is only a PR, has not been pushed to develop branch yet.  you might want to hold off on merging it to master at this time.

zaro0508 commented:
ohh actually i see that it's already been merged to develop so nevermind.  it should be ok since the new eb-deploy is setup to run in a separate CI stage.

zaro0508 committed:
fix eb deploy

diegotoptal committed:
fixes

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  fix eb deploy
  setup CD to AWS Beanstalk

zaro0508 committed:
pip does not have -y flag

diegotoptal committed:
remove team members from gene details

diegotoptal committed:
remove beta, bump version

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  pip does not have -y flag

diegotoptal committed:
drugabillity only visible for nominated targets

zaro0508 committed:
eb deploy command needs environment info

zaro0508 committed:
do not use awsebcli to deploy

Zaknarfen committed:
Fixing the charts error for AoT production build

diegotoptal committed:
about page and nominated target modal

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Fixing the charts error for AoT production build
  eb deploy command needs environment info

diegotoptal committed:
fixes

diegotoptal committed:
fix for non nominated menu

diegotoptal committed:
replace all #1A1C29 to variables with #24334F

zaro0508 committed:
Use travis eb provider for deployment

Use the Travis EB provider instead of awsebcli to deploy app to beanstalk.
The awsebcli required setting up a config file that was too onerous to
get working.

Zaknarfen committed:
Quick hotfix

Zaknarfen committed:
Adding async await to try and prevent chart bug

diegotoptal committed:
fix color to variable

diegotoptal committed:
Merge branches 'develop' and 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Adding async await to try and prevent chart bug
  Quick hotfix
  Use travis eb provider for deployment
  do not use awsebcli to deploy

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Adding async await to try and prevent chart bug
  Quick hotfix
  Use travis eb provider for deployment
  do not use awsebcli to deploy

diegotoptal committed:
fix nominated target min view

diegotoptal committed:
modal title color

diegotoptal committed:
multiple red dot call avoidance test

Zaknarfen committed:
Chart fixes for box-plot

Zaknarfen committed:
Another quick fix for the chart

Zaknarfen committed:
Added missing env variable to node scripts

Zaknarfen committed:
One more attempt at fixing the charts

diegotoptal committed:
remove beta from title

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  One more attempt at fixing the charts
  Added missing env variable to node scripts
  Another quick fix for the chart
  Chart fixes for box-plot

diegotoptal committed:
fix hide druggability not evidence on non nominated target

diegotoptal committed:
space between evidence menu links

zaro0508 committed:
disable tests for node ver 6 and 7

zaro0508 committed:
Merge pull request #544 from zaro0508/update-test

disable tests for node ver 6 and 7

diegotoptal committed:
remove druggabilty from SOE

diegotoptal committed:
Merge branches 'develop' and 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  disable tests for node ver 6 and 7

* 'develop' of github.com:Sage-Bionetworks/Agora:
  disable tests for node ver 6 and 7

diegotoptal committed:
fix drugability modal

diegotoptal committed:
fix sp dialog text

diegotoptal committed:
fix title test

Zaknarfen committed:
Removing tissue dropdown, changing box-plot

zaro0508 committed:
conditionally build client according to environment

Zaknarfen committed:
Merge pull request #551 from zaro0508/conditional-build

conditionally build client according to environment

Zaknarfen committed:
Removing compress from aot build

zaro0508 committed:
Revert "conditionally build client according to environment"

This caused invalid yaml exception.
This reverts commit a0c6a78f4c506f9ba52c562ac5989ac4202c5351.

Zaknarfen committed:
Axis fixing attempt

diegotoptal committed:
testing logs

diegotoptal committed:
nominated details text length

zaro0508 committed:
Revert "Revert "conditionally build client according to environment""

This reverts commit 35dce3962180b6be6e6d3c8e56b9d21483561154.

zaro0508 committed:
fix conditional build on beanstalk ec2

zaro0508 committed:
Merge pull request #552 from zaro0508/env-builds

conditional environment builds of agora

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  fix conditional build on beanstalk ec2
  Revert "Revert "conditionally build client according to environment""

Zaknarfen committed:
Small fix removing linebreak from local download script

diegotoptal committed:
filter Statistical and modal fix

diegotoptal committed:
Merge branch 'develop' of github.com:Sage-Bionetworks/Agora into develop

* 'develop' of github.com:Sage-Bionetworks/Agora:
  Small fix removing linebreak from local download script

diegotoptal committed:
Changed RNA header to h1 so difference is noticeable

diegotoptal committed:
narrow width of diff expr widget

Zaknarfen committed:
Adding new tooltip, fixing headings for new chart

	- Added X axis labels to the box plot
	- Added a new tooltip to the brain tissues
	in the X axis on the forest plot
	- Added the model above the forest-plot
	- Closes #556

diegotoptal committed:
fixes

Zaknarfen committed:
Removing duplicated dom elements

	- Removed the duplicated charts
	- Removed duplicated tooltips
	- Closes #558

diegotoptal committed:
nominated target modal fix

diegotoptal committed:
text update differential expression

diegotoptal committed:
updated legend network widget

diegotoptal committed:
missing space fixes

zaro0508 committed:
Attempt to fix nginx retart error

AWS support says that the script is trying to delete a file that
that doesn't exist.  Their suggestion is to ddd some logic to only delete
if file exists.

zaro0508 committed:
Merge pull request #562 from zaro0508/nginx-restart-fix

Attempt to fix nginx retart error

zaro0508 committed:
another attempt to fix nginx restart

This time AWS support suggests using initctl instead of service
command.

FROM AWS SUPPORT:
 I have done some research on the issue and have found that it is an
 ongoing issue with the node.js environment and the cause for the issue
 all node.js environment greater than equal to 4.4.6 version is
 reporting an error in service nginx restart.  I have found that instead
 of service command they are now using the initctl for starting and
 stopping the nginx service.

zaro0508 committed:
Merge pull request #564 from zaro0508/nginx-restart-fix2

another attempt to fix nginx restart

zaro0508 committed:
another attempt at nginx restart fix

command arguements need to be in correct order

zaro0508 committed:
Merge pull request #565 from zaro0508/nginx-restart-fix3

another attempt at nginx restart fix

Zaknarfen committed:
Small UI fixes

	- Changed embed video from YouTube to Vimeo
	- Updated Differential Expression page layout
	to match Figma design
	- Closes #541
	- Closes #557
